cachito/workers/pkg_managers/rubygems.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import re
+from dataclasses import dataclass
+from pathlib import Path
+
+from gemlock_parser.gemfile_lock import GemfileLockParser
+
+from cachito.errors import ValidationError
+
+GIT_REF_FORMAT = re.compile(r"^[a-fA-F0-9]{40}$")
+PLATFORMS_RUBY = re.compile(r"^PLATFORMS\n {2}ruby\n\n", re.MULTILINE)
+
+
+@dataclass
+class GemMetadata:
+    """Gem metadata."""
+
+    name: str
+    version: str
+    type: str
+    path: str
```

Comment:
IMO `path` is a slightly misleading name, makes sense for PATH gems but not the other types. Maybe use `remote`, like gemlock_parser does? Or `source`, which sounds applicable to both paths and urls

Reply:
Hmm, my point was to unite it, because the parser actually internally copies `remote` to `path` in case of PATH dependency, but maybe it wasn't the best idea. I'll create a new field to make naming clearer.